sdb/commands/linux/stacks.py

```diff
@@ -455,3 +455,129 @@ def pretty_print(self, objs: Iterable[drgn.Object]) -> None:
     def no_input(self) -> Iterable[drgn.Object]:
         self.validate_context()
         yield from [sdb.get_prog().crashed_thread().object]
+
+class ArgStack(sdb.Locator, sdb.PrettyPrinter):
+    """
+    Print a stack with arguments.
+
+    EXAMPLES
+    sdb> echo 0xffff96609b7d4680 | stack_args
+    (struct task_struct *)0xffff96609b7d4680
+    #0  struct rq *context_switch(struct rq *rq, struct task_struct *prev, struct task_struct *next, struct rq_flags *rf)
+          rf = (struct rq_flags *)<absent>
+          next = (struct task_struct *)<absent>
+          prev = (struct task_struct *)<absent>
+          rq = (struct rq *)<absent>
+    #1  void __schedule(bool preempt)
+          preempt = (bool)<absent>
+          prev = (struct task_struct *)0xffff96609b7d4680
+          next = (struct task_struct *)<absent>
+          switch_count = (unsigned long *)<absent>
+          rf = (struct rq_flags){ 0, {}, 4 }
+          rq = (struct rq *)<absent>
+          cpu = (int)<absent>
+    #2  void schedule()
+          tsk = (struct task_struct *)<absent>
+    ...
+    """
+
+    names = ["stack_args"]
+    input_type = "struct task_struct *"
+    output_type = "struct task_struct *"
+    load_on = [sdb.Kernel()]
+
+    def __init__(self,
+                 args: Optional[List[str]] = None,
+                 name: str = "_") -> None:
+        super().__init__(args, name)
+
+    @classmethod
+    def _init_parser(cls, name: str) -> argparse.ArgumentParser:
+        parser = super()._init_parser(name)
+        parser.add_argument(
+            "-r",
+            "--registers",
+            action="store_true",
+            help="print register values in output")
+        parser.add_argument(
+            "-f",
+            "--frame",
+            help="print detailed info for locals in a specific FRAME")
+        return parser
+
+    def validate_context(self) -> None:
+        if sdb.get_target_flags() & drgn.ProgramFlags.IS_LINUX_KERNEL == 0:
+            raise sdb.CommandError(self.name,
+                                   "command only works for linux kernel")
+
+    def pretty_print(self, objs: Iterable[drgn.Object]) -> None:
+        # self.validate_context()
+        for thread in objs:
+            # if thread.type_ != self.input_type:
+                #raise sdb.CommandError(
+                #    self.name, "can only print args for stacks")
+
+            if self.args.frame:
+                frame = sdb.get_prog().stack_trace(thread)[int(self.args.frame)]
```

Comment:
What happens if they specify a frame number that isn't valid (i.e. more than the number of frames, or non-numeric)?